### PR TITLE
5177 change echo and print in examples

### PR DESCRIPTION
**Pull request by snoek09**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | #5177 |

Original PR and discussion in https://github.com/symfony/symfony-docs/pull/5285

Change echo/print in examples to use:
- `var_dump()` in components docs
- `dump()` in framework docs

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@snoek09 thanks for proposing this improvement for the Symfony docs.

I don't know if this has already been discussed by the doc maintainers (@weaverryan, @WouterJ and @xabbuh) but I have some concerns about this change.

The problem is that we're changing something that works and everybody knows (`echo`, `print`) by something specific to Symfony and which probably won't work when you copy+paste it. In the book we could use it because you're probably using the full Symfony framework with the VarDumper component enabled. But for components documentation, the reader is probably not using the VarDumper component, so the examples will no longer work.

**Comment by snoek09** [comment]
To be clear, and improve discussion on this issue, @iltar proposed this change in #5177

**Comment by harikt** [comment]
> But for components documentation, the reader is probably not using the VarDumper component, so the examples will no longer work.

Good call. But in #5177 it mention var_(dump) . So as in components either dump should be changed to make use of  `var_dump` . Else the easy use of copy / paste will give a pain for new comers.

First impression will be dump function not available .

**Comment by xabbuh** [comment]
I'm actually not sure if we should use echo/print/dump/var_dump in most cases at all. Having an explanatory comment might be better in most cases.

**Comment by weaverryan** [comment]
This is a follow-up after #5285 - in there, we discussed using `dump()` in the framework (which is just a few spots) and then `var_dump()` in the components. @snoek09 can you change to use `var_dump()` in the components?

**Comment by snoek09** [comment]
@weaverryan I will commit the change to 'var_dump' this week.

File: book/doctrine.rst [hunk 1]
```diff
@@ -1204,7 +1204,8 @@ to the given ``Category`` object via their ``category_id`` value.
         $category = $product->getCategory();
 
         // prints "Proxies\AppBundleEntityCategoryProxy"
-        echo get_class($category);
+        dump(get_class($category));
+        die();
```
Comment (WouterJ) [inline book/doctrine.rst]: is there a reason why we want to have `die()`?

Comment (snoek09) [inline book/doctrine.rst]: From the original PR https://github.com/symfony/symfony-docs/pull/5285

`In the framework, dump() needs to be used with a die(), else it won't show up (it'll show up in the WDT).`

**Comment by snoek09** [comment]
@weaverryan `var_dump()` is in components.

**Comment by weaverryan** [comment]
I don't feel super-compelled that the change was needed in all places, but this has a :+1: from me

File: components/expression_language/syntax.rst [hunk 2]
```diff
@@ -140,14 +140,14 @@ Arithmetic Operators
 
 For example::
 
-    echo $language->evaluate(
+    dump($language->evaluate(
```
Comment (xabbuh) [inline components/expression_language/syntax.rst]: Shouldn't this be `var_dump()`?

Comment (snoek09) [inline components/expression_language/syntax.rst]: :+1: good catch!

**Comment by xabbuh** [comment]
:+1: Imo this is good to be merged.

**Comment by WouterJ** [comment]
Thanks for doing all these changes and keeping everything in sync for over a year. I've now merged this.

As 2.3 is the oldest version that we still maintain, I've split your PR into 2 PRs: #5451 (containing all fixes for 2.3) and #5452 (containing all added stuff in 2.6). These PRs use your original commits, so you still get all the credits you deserve.
As you see, I've also removed the merge commits from these PRs. These 2 PRs are now merged, so this PR is now fully merged (if I didn't do anything wrong).

A Git tip for the next time: Instead of using `git merge` to make your PR up to date with the original branches, use `git rebase`. This command rebases your commits on top of the latest version of the original branch. This means you don't get these not-so-nice merge commits in your PR. For a great explanation of `git rebase` vs `git merge`, please read https://www.atlassian.com/git/tutorials/merging-vs-rebasing/

**Comment by snoek09** [comment]
@WouterJ shouldn't it have been the other way around as `dump()` was added in 2.6?

Thanks for the git tip. I looked over the `git rebase` comment in the contributing docs.

**Comment by WouterJ** [comment]
Doh! I knew there had to be something why nobody noticed that 2.6 was not correct before me... Thanks for the heads-up.

I've changed all `dump()` usages to `var_dump()` in https://github.com/symfony/symfony-docs/pull/5454 , merged that in the 2.3 branch and reverted this change in the 2.6 branch in https://github.com/symfony/symfony-docs/commit/58ca2097b41ce7951cfe9cce060fec8aaaab312f Everything should be correct now :)

**Pull request closed without merging**